core/src/main/java/io/nixer/nixerplugin/core/detection/DetectionConfiguration.java

```diff
@@ -152,4 +153,42 @@ public UserAgentOverLoginThresholdRegistry userAgentRegistry() {
         }
     }
 
+    @Configuration
+    @ConditionalOnProperty(prefix = "nixer.rules.failed-login-ratio-level", name = "enabled", havingValue = "true")
+    static class FailedLoginRatioRule {
+
+        @Autowired
+        DetectionConfiguration detection;
+
+        @Autowired
+        NowSource nowSource;
+
+        @Bean
+        @ConfigurationProperties(prefix = "nixer.rules.failed-login-ratio-level")
+        public FailedLoginRatioProperties failedLoginRatioProperties() {
+            return new FailedLoginRatioProperties();
+        }
+
+        @Bean
+        public LoginRule failedLoginRatioRule() {
+            final FailedLoginRatioProperties properties = failedLoginRatioProperties();
+
+            return detection.ruleFactory().createFailedLoginRatioRule(
+                    properties.getWindow(),
+                    properties.getActivationLevel(),
+                    properties.getDeactivationLevel(),
+                    properties.getMinimumSampleSize());
+        }
+
+        @Bean
+        public FailedLoginRatioFilter failedLoginRatioFilter() {
+            return new FailedLoginRatioFilter(failedLoginRatioRegistry());
+        }
+
+        @Bean
+        public FailedLoginRatioRegistry failedLoginRatioRegistry() {
+            return new FailedLoginRatioRegistry(Duration.ofMinutes(20), nowSource);
```

Comment:
The duration value should be taken from `FailedLoginRatioProperties` and the default value of 20min should rather be defined as a constant.

Reply:
I didn't put that parameter in the configuration intentionally, it does not impact system behavior in a meaningful way and I wanted to keep configuration to contain only important parameters.